egs/cifar/v1/image/ocr/process_augment_data.py

```diff
@@ -11,68 +11,54 @@
 
 import os
 import argparse
-
 parser = argparse.ArgumentParser(description="""Regenerate images.scp, text,
                     utt2spk and spk2utt from feats.scp for augment data.""")
 parser.add_argument('dir', type=str,
                     help='directory of images.scp')
 args = parser.parse_args()
 
 text_file = os.path.join(args.dir, 'backup', 'text')
-text_dict = dict()  # stores image_id and text
-
+uttid_to_text = dict()  #stores uttid and text
 with open(text_file) as text_fh:
     for uttid_text in text_fh:
-        uttid_text = uttid_text.strip()
-        uttid_text_vect = uttid_text.split(" ")
-        uttid = uttid_text_vect[0]
-        image_id = uttid
+        uttid_text_vect = uttid_text.strip().split(" ")
+        utt_id = uttid_text_vect[0]
         text_vect = uttid_text_vect[1:]
         text = " ".join(text_vect)
-        text_dict[image_id] = text
+        uttid_to_text[utt_id] = text
 
 utt2spk_file = os.path.join(args.dir, 'backup', 'utt2spk')
-uttid_to_spk = dict()  # stores image_id and speaker
-
+uttid_to_spk = dict()  #stores uttid and speaker
 with open(utt2spk_file) as utt2spk_fh:
     for uttid_spk in utt2spk_fh:
-        uttid_spk = uttID_spk.strip()
-        uttid_spk_vect = uttid_spk.split(" ")
-        uttid = uttid_spk_vect[0]
-        image_id = uttid
+        uttid_spk_vect = uttid_spk.strip().split(" ")
+        utt_id = uttid_spk_vect[0]
         spk = uttid_spk_vect[1]
-        uttid_to_spk[image_id] = spk
+        uttid_to_spk[utt_id] = spk
 
 image_file = os.path.join(args.dir, 'backup', 'images.scp')
```

Comment:
Ah! I see now what you are doing. You make a copy of the data dir (as a backup), then call `make_features.py` to generate feats.scp with 2 times as many utterances as there are. Then you call this script to update text, images.scp and utt2spk to cover all the entries in feats.scp. Is that right?

I think it would be better to take this approach instead: remove anything related to augmentation in `extract_features.sh` add a make a new script `local/augment_data.sh <src-dir> <out-dir>` which does something like this:
```
utils/copy_data_dir.sh --spk-prefix aug1- --utt-prefix aug1- $srcdir ${outdir}_aug1
# call extract_features.sh on ${outdir}_aug1 with appropriate augmentation options (which will simply be passed to make_features.py)
# ... do as many augmentations as needed, e.g. aug2, aug3, ... and then combine them all:
utils/combine_data_dir.sh --extra-files images.scp $outdir  $srcdir ${outdir}_aug1 ${outdir}_aug2 ...
# remove the temporary augX dir's:
rm -r ${outdir}_aug1 ${outdir}_aug2 ...
```
You might need to modify `utils/copy_data_dir.sh` to be able to handle `images.scp`.

Reply:
Thank you. Yeah, the process_augment_data script is doing exactly same as you said (updating the text, images.scp and utt2spk to cover all the entries in feats.scp). I will surely make the above mentioned modifications for augmenting.